generators/server/templates/src/main/java/package/repository/_UserRepository.java

```diff
@@ -41,9 +39,11 @@
     Optional<User> findOneByLogin(String login);
     <%_ if (databaseType == 'sql') { _%>
 
-    @Query(value = "select distinct user from User user left join fetch user.authorities",
-        countQuery = "select count(user) from User user")
-    Page<User> findAllWithAuthorities(Pageable pageable);
+    @EntityGraph(attributePaths = "authorities")
+    User findOneWithAuthoritiesById(<%= pkType %> id);
+
+    @EntityGraph(attributePaths = "authorities")
+    Optional<User> findOneWithAuthoritiesByLogin(String login);
```

Comment:
@hdurix, do you have any idea how to make it work? Because therefore there is no sense in `@Cache` annotations on classes. I would even say that cache becomes useless in that case.
@ctamisier, I think the better solution is to find a way to get entities by primary key. Query cache is using on frequently loaded entities.